gk/main.c

```diff
@@ -2386,13 +2388,25 @@ gk_stage1(void *arg)
 	if (ret < 0)
 		goto cleanup;
 
+	gk_conf->num_mbuf = calculate_mempool_config_para("gk", gk_conf->net,
+		gk_conf->net->front.total_pkt_burst + (gk_conf->net->front.rss ?
```

Comment:
I don't understand why the test `gk_conf->net->front.rss` is correct. Can't RSS be disabled and blocks still have access to hardware filters?

Reply:
In fact, I don't understand why we need the field `INTERFACE.total_pkt_burst_hw_filter` to start with. Aren't the mbuf pools of the blocks themselves responsible for these packets?

Reply:
In our current implementation, the hardware filters can only be used if supported by the NIC (to steer matching packets) and if RSS is supported (to steer non-matching packets elsewhere). For GGU, CPS, and LLS, the blocks will only get their own RX queues if RSS is enabled, so we have the  test ``gk_conf->net->front.rss`` here.

Reply:
For the field ``INTERFACE.total_pkt_burst_hw_filter``, when the hw filters are supported, the mbuf pools of the blocks themselves are responsible for these packets, this field is not used. However, when the NIC doesn't have these filters, the packets will be directed from GK/GT blocks, thus the mempool of the GK/GT blocks will be responsible for these packets, in this case this field is used.